build.sbt

```diff
@@ -200,236 +231,92 @@ lazy val circe = project.in(file("."))
   .settings(
     libraryDependencies ++= Seq(
       "org.scalacheck" %%% "scalacheck" % scalaCheckVersion % "test",
       "org.scalatest" %%% "scalatest" % scalaTestVersion % "test"
     )
   )
-  .jvmSettings(
-    mimaPreviousArtifacts := Set("io.circe" %% "circe-numbers" % previousCirceVersion)
-  )
-  .jvmConfigure(_.copy(id = "numbers"))
-  .jsConfigure(_.copy(id = "numbersJS"))
 
 lazy val numbers = numbersBase.jvm
 lazy val numbersJS = numbersBase.js
 
-lazy val coreBase = crossProject.in(file("modules/core"))
-  .settings(
-    description := "circe core",
-    moduleName := "circe-core",
-    name := "Circe core",
-    crossScalaVersions := scalaVersions
-  )
-  .settings(allSettings: _*)
+lazy val coreBase = circeCrossModule("core", mima = previousCirceVersion)
   .settings(
-    libraryDependencies ++= Seq(
-      "org.typelevel" %%% "cats-core" % catsVersion
-    ),
+    libraryDependencies += "org.typelevel" %%% "cats-core" % catsVersion,
     sourceGenerators in Compile += (sourceManaged in Compile).map(Boilerplate.gen).taskValue
   )
-  .jvmSettings(
-    mimaPreviousArtifacts := Set("io.circe" %% "circe-core" % previousCirceVersion)
-  )
-  .jvmConfigure(_.copy(id = "core"))
-  .jsConfigure(_.copy(id = "coreJS"))
   .dependsOn(numbersBase)
 
 lazy val core = coreBase.jvm
 lazy val coreJS = coreBase.js
 
-lazy val genericBase = crossProject.in(file("modules/generic"))
-  .settings(
-    description := "circe generic",
-    moduleName := "circe-generic",
-    name := "Circe generic",
-    crossScalaVersions := scalaVersions
-  )
-  .settings(allSettings: _*)
-  .settings(macroDependencies: _*)
+lazy val genericBase = circeCrossModule("generic", mima = previousCirceVersion)
+  .settings(macroDependencies)
   .settings(
-    libraryDependencies += "com.chuusai" %%% "shapeless" % shapelessVersion,
-    sources in (Compile, doc) := (
-      CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, 11)) => (sources in (Compile, doc)).value
-        case _ => Nil
-      }
-    )
-  )
-  .jvmSettings(
-    mimaPreviousArtifacts := Set("io.circe" %% "circe-generic" % previousCirceVersion)
+    libraryDependencies += "com.chuusai" %%% "shapeless" % shapelessVersion
   )
-  .jvmConfigure(_.copy(id = "generic"))
-  .jsConfigure(_.copy(id = "genericJS"))
   .dependsOn(coreBase)
 
 lazy val generic = genericBase.jvm
 lazy val genericJS = genericBase.js
 
-lazy val genericExtrasBase = crossProject.crossType(CrossType.Pure).in(file("modules/generic-extras"))
-  .settings(
-    description := "circe generic extras",
-    moduleName := "circe-generic-extras",
-    name := "Circe generic extras",
-    crossScalaVersions := scalaVersions
-  )
-  .settings(allSettings: _*)
-  .settings(macroDependencies: _*)
-  .settings(
-    sources in (Compile, doc) := (
-      CrossVersion.partialVersion(scalaVersion.value) match {
-        case Some((2, 11)) => (sources in (Compile, doc)).value
```

Comment:
If I remember correctly this is about the non-Unidoc doc generation that happens automatically during publishing.

The problem is that it chokes on some of the macros here (I believe the annotation macros) on 2.10, and I was avoiding this by excluding the sources from the Javadoc artifact generation on 2.10.

You're right that 2.12 should have been added here, but the fact that it wasn't just resulted in Javadoc artifacts being empty for 2.10 _and_ 2.12:

* https://repo1.maven.org/maven2/io/circe/circe-generic_2.10/0.6.1/
* https://repo1.maven.org/maven2/io/circe/circe-generic_2.11/0.6.1/
* https://repo1.maven.org/maven2/io/circe/circe-generic_2.12/0.6.1/

This isn't a big deal, but removing the removal entirely will cause 2.10 publishing to break.

Ideally we should reinstate the removal for 2.10 only, so that 2.11 and 2.12 have proper Javadoc artifacts.